src/boot.c

```diff
@@ -446,30 +454,120 @@ static void check_fat_state_bit(DOS_FS * fs, void *b)
     }
 }
 
+static int is_fat32(const struct boot_sector *b)
+{
+    return !b->fat_length && b->fat32_length;
+}
+
+static int try_boot_sector(off_t *byte_offset, unsigned offset,
+                           struct boot_sector *b)
+{
+    unsigned int logical_sector_size;
+
+    *byte_offset = offset * SECTOR_SIZE;
+    fs_read(*byte_offset, sizeof(*b), b);
+    logical_sector_size = GET_UNALIGNED_W(b->sector_size);
+
+    return
+	logical_sector_size &&
+	/* ...else logical sector size is zero. */
+
+	/* This is the first thing that will fail if the platform needs special
+	 * handling of unaligned multibyte accesses but such handling isn't
+	 * being provided. See GET_UNALIGNED_W() above. */
+	!(logical_sector_size & (SECTOR_SIZE - 1)) &&
+	/* ...else logical sector size is not a multiple of the physical sector
+	 * size. */
+
+	b->cluster_size &&
+	/* ...else cluster size is zero. */
+
+	b->fats &&
+	/* ...else number of FATs is zero. (NTFS is like this.) */
+
+	!(*byte_offset % logical_sector_size);
+	/* ...else the boot sector isn't on a logical sector boundary. */
+}
+
+static int has_signature(const void *sector)
+{
+    return le16toh(((const uint16_t *)sector)[255]) == 0xaa55;
```

Comment:
... or you can directly check `sector[510] == 0xFF && sector[511] == 0xAA`, make `sector` of type `unsigned char*` (or `uint8_t*`) to prevent problems with endianity...

Reply:
Sure, why not.

(Though I think my way would be around *a whole nanosecond* faster!)